js/private/node-patches/src/fs.ts

```diff
@@ -621,19 +642,16 @@ export const patcher = (fs: any = _fs, roots: string[]) => {
         function _tryParent(): boolean {
             nested.push(path.basename(maybe))
             const dirname = path.dirname(maybe)
-            if (
-                !dirname ||
-                dirname == maybe ||
-                dirname == '.' ||
-                dirname == '/'
-            ) {
+            // TODO: also look for paths that are not within any 'roots'
+            if (dirname == maybe || HOP_ROOTS[dirname] || HOP_ROOTS[maybe]) {
                 // not a link
                 return false
             }
             maybe = dirname
             return true
         }
         for (;;) {
+            // TODO: cache anything within 'roots'
```

Comment:
Simply caching everything (within 'roots' which we know is readonly) causes test failures as well that I don't understand. Maybe some npm modules write to their own directory? It was causing runtime issues with I think `require` statements failing.